src/rational/mat_q/sample/gauss.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright © 2024 Marvin Beckmann
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains sampling algorithms for gaussian distributions over [`MatQ`].
+
+use crate::{
+    error::MathError,
+    rational::{MatQ, Q},
+    traits::{GetNumColumns, GetNumRows, SetEntry},
+};
+use std::fmt::Display;
+
+impl MatQ {
+    /// Chooses a [`MatQ`] instance according to the continuous Gaussian distribution.
+    /// Here, each entry is chosen according to the provided distribution.
+    ///
+    /// Parameters:
+    /// - `num_rows`: specifies the number of rows the new matrix should have
+    /// - `num_cols`: specifies the number of columns the new matrix should have
+    /// - `center`: specifies the position of the center
+    /// - `sigma`: specifies the standard deviation
+    ///
+    /// Returns new [`MatQ`] sample chosen according to the specified continuous Gaussian
+    /// distribution or a [`MathError`] if the specified parameters were not chosen
+    /// appropriately, `sigma > 0`.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    ///
+    /// let sample = MatQ::sample_gauss(5, 5, 0, 1).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`NonPositive`](MathError::NonPositive)
+    ///     if `sigma <= 0`.
+    ///
+    /// # Panics ...
+    /// - if the provided number of rows and columns are not suited to create a matrix.
+    ///     For further information see [`MatQ::new`].
+    pub fn sample_gauss(
+        num_rows: impl TryInto<i64> + Display,
+        num_cols: impl TryInto<i64> + Display,
+        center: impl Into<f64>,
```

Comment:
In Q, you had 'center: impl Into\<Q\>', why is it here 'Into\<f64\>'?
Maybe change it to 'Into\<Q\>'.

Reply:
reworked this to take in a MatQ instead which indirectly provides the dimensions for the sample -> please check again